app/controllers/healthcheck_controller.rb

```diff
@@ -2,11 +2,13 @@ class HealthcheckController < ApplicationController
   skip_before_action :authenticate_user!
 
   def index
-    healthcheck = GovukHealthcheck.healthcheck([
+    checks = [
       GovukHealthcheck::SidekiqRedis,
       GovukHealthcheck::ActiveRecord,
       Healthcheck::GovernmentDataCheck,
-    ])
-    render json: healthcheck
+    ]
+
+    checks << Healthcheck::ActiveStorage if params[:storage]
```

Comment:
What's the purpose of the param here?

Reply:
Oh, maybe it's not clear from the test/commit: this PR is a suggestion for how we could implement an optional check for S3 connectivity as part of the existing `/healthcheck` endpoint.

I've for this in the commit, and against on the PR description, which I guess is a bit confusing! Hopefully it makes sense now, though. I've made this a draft PR, and will tidy up if we reach an agreement on the approach.

Reply:
Thanks, it didn't really answer my query I'm afraid. I'm a bit confused why this would be something optional as ActiveStorage should work in all environments (even those without S3) so I don't know why we'd need it to be optional. Maybe this concern is different for other apps that directly communicate with S3?

Reply:
Ah, so this is because there are multiple uses of the `/healthcheck` endpoint:

- LB healthcheck, which seems to be roughly every 10s, per machine.
- Alert healthcheck (every 5mins, per machine).
- (New) Smokey healthcheck, as part of enabling CD.

We need to add the check as part of enabling CD, but it's not clear to me if we should use it as a signal for the app being unavailable, which how the healthcheck gets interpreted when requested by the LB.

So I've suggested a compromise of making it an optional check on the endpoint. In the description I've argued why it could also be just like any other healthcheck.

Does that make more sense? (I can update the PR/commit to explain better.)

Reply:
Ok thanks, I think I understand. 

I'm not particularly keen on params to turn on different aspects of the healthcheck if we don't want to care about whether they pass or not in some situations. This feels to me like a conflation of the three states of healthcheck: ok, warning and critical. It feels like if we're not sure we care about this then it should be a warning rather than a critical?

Reply:
Fair enough - I wasn't set on the approach, and was also considering a WARNING status. Given the problems with the description and approach, I've raised [a fresh PR](https://github.com/alphagov/content-publisher/pull/2204) with that approach.

Could you take a look at that one?